src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -420,12 +428,18 @@ FCIMPL3(void, WeakReferenceNative::Create, WeakReferenceObject * pThisUNSAFE, Ob
 
     // Create the handle.
 #ifdef FEATURE_COMINTEROP
-    IWeakReference* pRawWinRTWeakReference = GetWinRTWeakReference(&gc.pTarget);
-    if (pRawWinRTWeakReference != nullptr)
+    IWeakReference* pRawComWeakReference = nullptr;
+    SyncBlock* pSyncBlock = gc.pTarget != NULL ? gc.pTarget->PassiveGetSyncBlock() : nullptr;
+    if (gc.pTarget != NULL && pSyncBlock && pSyncBlock->GetInteropInfoNoCreate() != nullptr)
```

Comment:
`gc.pTarget != NULL &&` is unnecessary.

It may be easier to read if written this way:
```
if (gc.pTarget != NULL)
{
    SyncBlock* pSyncBlock = gc.pTarget->PassiveGetSyncBlock();
    if (pSyncBlock && pSyncBlock->GetInteropInfoNoCreate() != nullptr)
    {
...
```

Reply:
Lets also be consistent about `nullptr` vs `NULL`. I have tried to match the more common usage of `NULL` in existing CLR files. New files I like to adopt the C++11 approach.

Reply:
Currently `OBJECTREF` doesn't support being assigned or compared to a `std::nullptr_t`, so that's why I've been using `NULL` for `gc.pTarget`.